### ISSUE TITLE
`rabbitmq_queue_exchange_messages_published_total` in `/metrics/per-object` has a duplicate `vhost` label

### ISSUE THREAD

**Issue by Gama11** [rabbitmq/rabbitmq-server#12347]
### Describe the bug

Prometheus seems to choke on duplicate labels:

![image](https://github.com/user-attachments/assets/7ee431fd-1be0-47fa-a4e7-3220674225ee)

Note how `vhost` appears twice here:

```
# TYPE rabbitmq_queue_exchange_messages_published_total counter
# HELP rabbitmq_queue_exchange_messages_published_total Total number of messages published into a queue through an exchange
rabbitmq_queue_exchange_messages_published_total{vhost="/",queue="<queue-name>",vhost="/",exchange="<exchange-name>"} 19
```

Versions used:

- rabbitmq:4.0.0-management-alpine
- prom/prometheus:v2.54.1

This seems to be a regression compared to 3.13.7 which we upgraded from.

### Reproduction steps

1. Enable the `rabbitmq_prometheus` plugin
2. Send some messages to a queue so the `rabbitmq_queue_exchange_messages_published_total` metric is included
3. Query the `/metrics/per-object` endpoint

### Expected behavior

There should not be duplicate labels.

**Comment by Gama11** [comment]
Possibly relevant, our `rabbitmq.config`:

```erlang
[
  {
    rabbit,
      [
        { loopback_users, [] }
      ]
  },
  {
    rabbitmq_management,
      [
        { load_definitions, "/etc/rabbitmq/definitions.json" }
      ]
  }
].
```

`definitions.json`:

```json
{
  "users": [
    {
      "name": "test",
      "password_hash": "<redacted>",
      "hashing_algorithm": "rabbit_password_hashing_sha256",
      "tags": "administrator,monitoring"
    }
  ],
  "vhosts": [
    {
      "name": "/"
    }
  ],
  "permissions": [
    {
      "user": "test",
      "vhost": "/",
      "configure": ".*",
      "write": ".*",
      "read": ".*"
    }
  ],
  "topic_permissions": [],
  "parameters": [],
  "global_parameters": [],
  "policies": [],
  "queues": [],
  "exchanges": [],
  "bindings": []
}
```

**Comment by michaelklishin** [comment]
Thank you. This looks like a blank node's config with a pre-configured user for remote access.

I was able to reproduce with a blank node from `main`:

```
# HELP rabbitmq_queue_exchange_messages_published_total Total number of messages published into a queue through an exchange
rabbitmq_queue_exchange_messages_published_total{vhost="/",queue="qq.1",vhost="/",exchange=""} 100000
```

**Comment by gomoripeti** [comment]
I think this is related to https://github.com/rabbitmq/rabbitmq-server/pull/11559
the first vhost belongs to the queue, the second to the exchange but obviously they are the same so need to special handle this new case to eliminate the redundancy

**Comment by Gama11** [comment]
I noticed that `rabbitmq_queue_messages_published_total` has two labels called `exchange_vhost` and `queue_vhost` (or at least it did in 3.13.7).

**Comment by pschichtel** [comment]
is there ever a case where a queue's vhost and the exchange's vhost would be different?

**Comment by michaelklishin** [comment]
@pschichtel no, by definition, entities in different virtual hosts cannot be bound together.

**Comment by michaelklishin** [comment]
The labels may simply reflect the fact that there are two associated entities with certain metric samples, and they both belong to a virtual host. Ideally it should be just `vhost`, `virtual_host`, or whatever is the label we use for other metrics.

**Comment by michaelklishin** [comment]
A fix was contributed by @LoisSotoLopez in #12364, merged in #12373 and (will be) backported in #12378.
